### DISCUSSION TITLE
Not Able to deploy iTwinViewer Single page application to Heroku Platform.

### DISCUSSION THREAD

**Discussion by tejabennuri** [iTwin/itwinjs-core#4583] [Q&A]
Hello Everyone,
I want to deploy the viewer app to heroku platform. I created the app using:
` npx create-react-app your-app-name --template @itwin/web-viewer --scripts-version @bentley/react-scripts`
link: [reference document](https://www.itwinjs.org/learning/tutorials/develop-web-viewer/)
I was able to get the app running on localhost.
<img width="954" alt="Screenshot 2022-11-02 125158" src="https://user-images.githubusercontent.com/73427170/199494917-5d0e7f9f-a32b-45aa-8358-2f878f65691a.png">
I then proceeded to deploy application on Heroku Platform following the below article
link: [deploy iTwin app to Heroku](https://medium.com/itwinjs/deploying-the-itwin-viewer-to-a-web-host-d45c5cfdf0cf)
My deployment is failing with error showing node version not specified in package.json file
<img width="718" alt="Screenshot 2022-11-02 130201" src="https://user-images.githubusercontent.com/73427170/199496432-fe4e4d6d-fd25-45c4-a7c3-e3b785523549.png">

I have attached my log file of build here.
[mylog.txt](https://github.com/iTwin/itwinjs-core/files/9920420/mylog.txt)

**Comment by calebmshafer** [comment]
@tejabennuri based on the command you used the lock file will be a `yarn.lock` file instead of the `package-lock.json` it's looking for in the build.

I'd recommend either re-running the create-react-app command and add `--use-npm` or switch the Heroku build steps to use `yarn install` instead of `npm ci`.

**Reply by arnab-git-hub** [reply]
@tejabennuri    Does Caleb's answer resolve your issue? If yes, please mark the thread as answered.
